docs/developer-docs/setup/cycles/cycles-ledger.md

```diff
@@ -0,0 +1,158 @@
+
+### Checking your cycles balance
+
+To check the cycles balance of your user principal ID, use the command:
+
+```
+dfx cycles balance --network ic
+```
+
+### Converting ICP to cycles
+
+To convert ICP into cycles using the cycles ledger, use the command:
+
+```
+dfx cycles convert AMOUNT --network ic
+```
+
+Replace `AMOUNT` with the number of ICP to convert into cycles, such as `2.7`. Alternatively, the amount can be specified in terms of ICP and e8s (where `100,000,000 e8s = 1 ICP`) using the options `--icp` and `--e8s`. For example, `2.7` would be expressed as `--icp 2 --e8s 70_000_000`.
+
+### Transferring cycles to a principal
+
+To transfer cycles to another principal ID, use the command:
+
+```
+dfx cycles transfer AMOUNT PRINCIPAL_ID (--subaccount [SUBACCOUNT]) --network ic
+```
+
+Replace `AMOUNT` with the number of cycles, such as `34000000`, and `PRINCIPAL_ID` with the principal ID you'd like to transfer the cycles to, such as `tsqwz-udeik-5migd-ehrev-pvoqv-szx2g-akh5s-fkyqc-zy6q7-snav6-uqe`. Optionally, a subaccount can be provided, in which case the recipient will receive the funds in the provided subaccount.
+
+### Topping up a canister
+
+To transfer cycles into a canister to top it up, use the command:
+
+```
+dfx cycles top-up `AMOUNT` `CANISTER_ID` --network ic
+```
+
+Replace `AMOUNT` with the number of cycles, such as `34000000`, and `CANISTER_ID` with the canister ID you'd like to transfer the cycles to, such as `gastn-uqaaa-aaaae-aaafq-cai`.
+
+### Creating a canister using cycles
+
+New canisters can be created using the cycles ledger by using the `dfx canister create` and `dfx deploy` commands.
+
+Both of these commands have the following options and syntax to create a new canister:
+
+```
+dfx canister create CANISTER_NAME --network ic
+dfx deploy CANISTER_NAME --network ic
+```
+
+Replace `CANISTER_NAME` with a name for the new canister.
+
+With these two commands, there are two **optional** flags that can be used:
+
+- `dfx create canister CANISTER_NAME --next-to CANISTER_ID:` The new canister will be installed on the same subnet as the canister provided.
+- `dfx create canister CANISTER_NAME --subnet SUBNET_ID`: The new canister will be installed on the given subnet.
+
+If neither flag is used, the canister will be created on a random subnet using the user principal’s cycles.
+
+### Using ICP instead of cycles for canister creation
+
+Alternatively, if you want to use ICP instead of cycles for creating a canister, you can use the flag `--using-icp`, which will charge your principal ICP instead of cycles such as:
+
+```
+dfx create canister CANISTER_NAME --using-icp --network ic
+```
+
+Or you can set a specific amount of ICP with the option `--using-icp-amount AMOUNT`, such as:
+
+```
+dfx create canister CANISTER_NAME --using-icp-amount 2.7 --network ic
+```
```

Comment:
```suggestion
```

I have not implemented `--using-icp`

Reply:
Correct me if I'm wrong but isn't this "only" some rewiring and adding the flag? `dfx` currently uses ICP anyway to create canisters, right?
I'm basically asking what we should do with this part: Remove it for now until we have this functionality or only merge the documentation once the functionality is available?

Reply:
I think it is best to remove it for now, then update the documentation in the future once this functionality is implemented.